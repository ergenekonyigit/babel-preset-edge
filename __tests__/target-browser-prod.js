import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

describe("Target: Browser: Prod", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    target: "browser",
    env: "production"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
