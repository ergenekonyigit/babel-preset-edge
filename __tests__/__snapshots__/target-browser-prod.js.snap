// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Target: Browser: Prod comments 1`] = `"import _camelCase from\\"lodash/camelCase\\";function translate(a){return _camelCase(a)}"`;

exports[`Target: Browser: Prod deadcode 1`] = `"var STATIC=!1;!1,before(),!1,after();"`;

exports[`Target: Browser: Prod es2015: classes 1`] = `"import _inheritsLoose from\\"@babel/runtime/helpers/builtin/es6/inheritsLoose\\";export var Component=/*#__PURE__*/function(a){function b(){var b;return b=a.call(this)||this,b.flag=!0,b}var c=b.prototype;return c.main=function main(){alert(this.foo)},_inheritsLoose(b,a),b}(Other);"`;

exports[`Target: Browser: Prod es2015: for_off 1`] = `"import\\"core-js/modules/es7.symbol.async-iterator\\";import\\"core-js/modules/es6.symbol\\";import\\"core-js/modules/web.dom.iterable\\";function test(){for(var a=tasks,b=Array.isArray(a),c=0,a=b?a:a[Symbol.iterator]();;){var d;if(b){if(c>=a.length)break;d=a[c++]}else{if(c=a.next(),c.done)break;d=c.value}try{executeCommands()}catch(a){console.error(\\"Error\\")}}}"`;

exports[`Target: Browser: Prod es2015: generator 1`] = `"function*idMaker(){for(var a=0;;)yield a++}"`;

exports[`Target: Browser: Prod es2015: set 1`] = `"import\\"core-js/modules/es6.set\\";new Set([1,2,3]);"`;

exports[`Target: Browser: Prod es2016: exponentiation 1`] = `"var result=8;"`;

exports[`Target: Browser: Prod esnext: async 1`] = `"function wait(){return new Promise(function(a,b){return Promise.resolve(loadData()).then(function(){try{return a()}catch(a){return b(a)}},b)})}"`;

exports[`Target: Browser: Prod esnext: async_loop 1`] = `"import\\"core-js/modules/es7.symbol.async-iterator\\";import\\"core-js/modules/es6.symbol\\";import\\"core-js/modules/web.dom.iterable\\";function executeTasks(a){return new Promise(function(b,c){function d(){if(g){if(h>=f.length)return[1];i=f[h++]}else{if(h=f.next(),h.done)return[1];i=h.value}var a=function(){try{return d}catch(a){return c(a)}},b=function(){try{return console.error(\\"Error\\"),a()}catch(a){return c(a)}};try{return Promise.resolve(executeCommands()).then(function(){try{return a()}catch(a){return b(a)}},b)}catch(a){b(a)}}function e(){return b()}var f,g,h,i;f=a,g=Array.isArray(f),h=0,f=g?f:f[Symbol.iterator]();var j;return(j=function(a){for(;a;){if(a.then)return void a.then(j,c);try{if(a.pop){if(a.length)return a.pop()?e.call(this):a;a=d}else a=a.call(this)}catch(a){return c(a)}}}.bind(this))(d)})}"`;

exports[`Target: Browser: Prod esnext: class properties 1`] = `"import _defineProperty from\\"@babel/runtime/helpers/builtin/es6/defineProperty\\";export var Component=function(){_defineProperty(this,\\"onClick\\",function(){})};"`;

exports[`Target: Browser: Prod esnext: dynamic import 1`] = `"import(/*webpackChunkName:'lodash-pst7t'*/\\"lodash\\").then(function(a){console.log(a.camelCase(\\"hello world\\"))});"`;

exports[`Target: Browser: Prod esnext: object spread 1`] = `"import _objectSpread from\\"@babel/runtime/helpers/builtin/es6/objectSpread\\";var original={bar:10},variant=_objectSpread({foo:1},original);"`;

exports[`Target: Browser: Prod loadable 1`] = `"import loadable from\\"loadable-components\\";export var Home=loadable(function(){return import(/*webpackChunkName:'Home-bqRLr'*/\\"./views/Home\\")},{modules:[\\"./views/Home\\"]});export var About=loadable(function(){return import(/*webpackChunkName:'About-q5EMT'*/\\"./views/About\\")},{modules:[\\"./views/About\\"]});export var Contact=loadable(function(){return import(/*webpackChunkName:'Contact-Urv1P'*/\\"./views/Contact\\")},{modules:[\\"./views/Contact\\"]});"`;

exports[`Target: Browser: Prod loadable magic-comments 1`] = `"import loadable from\\"loadable-components\\";export var Home=loadable(function(){return import(/*webpackChunkName:'Home-bqRLr'*/\\"./views/Home\\")},{modules:[\\"./views/Home\\"]});export var About=loadable(function(){return import(/* webpackChunkName: \\"About\\" */\\"./views/About\\")},{modules:[\\"./views/About\\"]});export var Contact=loadable(function(){return import(/*webpackMode:'lazy',webpackFoo:'bar',webpackChunkName:'Contact-Urv1P'*/\\"./views/Contact\\")},{modules:[\\"./views/Contact\\"]});"`;

exports[`Target: Browser: Prod loadable wildcard 1`] = `"import loadable from\\"loadable-components\\";var page=\\"About\\",lang=\\"de_DE\\";export var views=loadable(function(){return import(/*webpackChunkName:'[request]-eNKDf'*/\\"./views/About\\")},{modules:[\\"./views/About\\"]});export var leanIntl=loadable(function(){return import(/*webpackChunkName:'[request]-zd7G4'*/\\"./lean-intl/\\"+lang+\\"/data\\")},{modules:[\\"./lean-intl/\\"+lang+\\"/data\\"]});export var reactIntl=loadable(function(){return import(/*webpackChunkName:'[request]-5gr63'*/\\"./react-intl/\\"+lang+\\"/data\\")},{modules:[\\"./react-intl/\\"+lang+\\"/data\\"]});"`;

exports[`Target: Browser: Prod lodash: async 1`] = `"import _parallel from\\"async/parallel\\";_parallel([]);"`;

exports[`Target: Browser: Prod lodash: lodash 1`] = `"import _camelCase from\\"lodash/camelCase\\";_camelCase(\\"hello world\\");"`;

exports[`Target: Browser: Prod lodash: recompose 1`] = `"import _compose from\\"recompose/compose\\";_compose([]);"`;

exports[`Target: Browser: Prod react: jsx 1`] = `"import _jsx from\\"@babel/runtime/helpers/builtin/es6/jsx\\";var _ref=/*#__PURE__*/_jsx(\\"h1\\",{},void 0,\\"Hello\\");function render(){return _ref}"`;

exports[`Target: Browser: Prod react: jsx spread 1`] = `"import\\"core-js/modules/es6.object.assign\\";function render(a){return React.createElement(\\"h1\\",Object.assign({className:\\"large\\"},a),\\"Hello\\")}"`;

exports[`Target: Browser: Prod react: proptypes 1`] = `"import _jsx from\\"@babel/runtime/helpers/builtin/es6/jsx\\";import _inheritsLoose from\\"@babel/runtime/helpers/builtin/es6/inheritsLoose\\";import React from\\"react\\";var MyComponent=/*#__PURE__*/function(a){function b(){return a.apply(this,arguments)||this}var c=b.prototype;return c.render=function render(){return _jsx(\\"div\\",{},void 0,this.props.userName)},_inheritsLoose(b,a),b}(React.Component);"`;
